the-collab-lab/tcl-12-smart-shopping-list
# Issue #10. Compute estimated next purchase date and record in the database

## Description

- Added 3 new items to Firestore
  - Number of Purchases
  - Old Purchased Date
  - Calculated Estimate of next purchase day using the estimates.js lib
- Used Dayjs to find the date difference

## Related Issue

Closes #10 

## Acceptance Criteria

- When a purchase is recorded, the estimated number of days until the next purchase date should be calculated and recorded in the database

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

![Screen Shot 2020-11-03 at 6 36 40 PM](https://user-images.githubusercontent.com/24923742/98052546-14088980-1dfc-11eb-973e-1400ddb7a443.png)



### After

After first purchase (notice calculatedEstimate):
![Screen Shot 2020-11-03 at 6 36 58 PM](https://user-images.githubusercontent.com/24923742/98052607-3dc1b080-1dfc-11eb-88e2-c962fb6159ad.png)


After 2+ purchases (notice calculatedEstimate):
![Screen Shot 2020-11-03 at 6 37 30 PM](https://user-images.githubusercontent.com/24923742/98052630-4dd99000-1dfc-11eb-9d9d-f2285763c471.png)


## Testing Steps / QA Criteria

- Switch to our branch `git checkout al-gj-compute-next-purchase`, then run `git pull origin al-gj-compute-next-purchase`. Start server with `npm start`.
- Join an existing list or create a new list and add a new item
- Open Firestore to see the information
- Notice your new item has only 3 fields. Now checkmark the item and notice 3 new fields pop up.
- Manually change the date in firestore to 24 hours before so your item unchecks. Now checkmark it again
- Notice the 3 new fields all update based on your last click to this click.
